packages/block-editor/src/hooks/custom-class-name.js

```diff
@@ -117,63 +112,6 @@ export function addSaveProps( extraProps, blockType, attributes ) {
 	return extraProps;
 }
 
-/**
- * Given an HTML string, returns an array of class names assigned to the root
- * element in the markup.
- *
- * @param {string} innerHTML Markup string from which to extract classes.
- *
- * @return {string[]} Array of class names assigned to the root element.
- */
-export function getHTMLRootElementClasses( innerHTML ) {
-	innerHTML = `<div data-custom-class-name>${ innerHTML }</div>`;
-
-	const parsed = parseWithAttributeSchema( innerHTML, {
-		type: 'string',
-		source: 'attribute',
-		selector: '[data-custom-class-name] > *',
-		attribute: 'class',
-	} );
-
-	return parsed ? parsed.trim().split( /\s+/ ) : [];
-}
-
-/**
- * Given a parsed set of block attributes, if the block supports custom class
- * names and an unknown class (per the block's serialization behavior) is
- * found, the unknown classes are treated as custom classes. This prevents the
- * block from being considered as invalid.
- *
- * @param {Object} blockAttributes Original block attributes.
- * @param {Object} blockType       Block type settings.
- * @param {string} innerHTML       Original block markup.
- *
- * @return {Object} Filtered block attributes.
- */
-export function addParsedDifference( blockAttributes, blockType, innerHTML ) {
```

Comment:
I'm finding the difference between, the "blocks" and the "block-editor" package more and more arbitrary or hard to define properly.

Removing this her means that the behavior of fixing the block className support is "built-in" into the blocks package and blocks parsing, meaning it's not really a "hook" anymore. 

On the other hand, all the built-in block editor hooks should probably be in `blocks` directly but we can't just do that because most of them rely on reusable components which would need to moved as well. 

Not sure what's the best solution for this in the future.

Reply:
The difference is that one of them is for "regristration of blocks" and "parsing/serializing" while the other is for the UI of the block editor.